text/0796-ember-data-deprecate-rsvp.md

```diff
@@ -0,0 +1,76 @@
+---
+Stage: Proposed
+Start Date: 2022-02-20
+Release Date: Unreleased
+Release Versions:
+  ember-source: vX.Y.Z
+  ember-data: vX.Y.Z
+Relevant Team(s): ember-data 
+RFC PR: https://github.com/emberjs/rfcs/pull/796
+---
+
+# Deprecate `RSVP.Promise` for native Promise
+
+## Summary
+
+All methods currently returning an `RSVP.Promise` will return a native Promise. In addition, all documentation will be updated accordingly.
+
+## Motivation
+
+With the removal of IE11 from our build infrastructure and browser requirements, we can now safely remove `RSVP.Promise`. From the [docs](https://github.com/tildeio/rsvp.js/):
+
+> Specifically, it is a tiny implementation of Promises/A+. It works in node and the browser (IE9+, all the popular evergreen ones).
+
+RSVP was Ember's polyfill since an early v1.0.0 version during a time when native Promises didn't exist in any browser.  In addition, Promises have been supported since Node 0.12.
+
+By removing `RSVP.Promise` in favor of native Promises, we can drop an unnecessary dependency for both client side and server side fetching of data.
+
+According to [bundlephobia](https://bundlephobia.com/package/rsvp@4.8.5), this would allow us to remove a significant chunk of dependency weight.
+
+## Detailed design
+
+Two steps will be required to fulfill deprecating `RSVP.Promise`.
+
+First, we will issue a deprecation `DEPRECATE_RSVP_PROMISE` to async callbacks that might be relying on a convenient feature of `RSVP.Promise`.  Namely, an `RSVP.Promise` may still be hanging by the time the underlying model instance or store has been destroyed.  This will help users surface instances where their test suite or code is dealing with dangling promises.  After the removal of this deprecation, this will throw an Error.
+
+- `id: ember-data:rsvp-promise-hanging`
+
+Second, we will also utilize the deprecation to replace all instances of `RSVP.Promise` with native Promises.
+
+The final implementation will look something like:
+
+```js
+  ajax(options) {
+    if (DEPRECATE_RSVP_PROMISE) {
+      return new Promise((resolve, reject) => {
+        ...
+      });
+    } else {
+      return new RSVPPromise((resolve, reject) => {
+        ...
+      });
+    }
+  },
+```
+
+## How we teach this
+
+We do not believe this requires any update to the Ember curriculum. API documentation may be needed to remove traces of `RSVP.Promise`.
```

Comment:
It has been my experience that many feel that RSVP offers two features not available in native Promises: `.hash()` and test waiters support (`await settled()` support).

If this is not that case—

1. Can we add to the "How we teach this" how to support `.hash()` or offer an alternative using native promises.
2. Can we add ti the "How do we teach this" an explanation why native promises are test-waiter compatible and that the integration that RSVP offered is no longer needed.

Reply:
@sukima

1) whether we return a native promise or an rsvp promise does not change anyone's ability to use `hash`.
2) good call out, it's unnecessary because the test waiters aren't waiting on RSVP Promises in this case either, they are waiting on various other things, which ember-data already installs waiters for.

Reply:
Oh! This RFC is for Ember-data  I thought it was for Ember proper  gotcha. :+1: